src/TCommandLine.cpp

```diff
@@ -672,192 +672,202 @@ void TCommandLine::slot_popupMenu()
-            *
-            * The insertAction[s](...)/(Separator(...)) insert their things
-            * second argument (or generated by themself) before the first (or
-            * only) argument given.
-            */
-
-            auto separator_aboveStandardMenu = popup->insertSeparator(popup->actions().first());
-            if (handle_profile) {
-                popup->insertAction(separator_aboveStandardMenu, action_removeWord);
-                popup->insertAction(action_removeWord, action_addWord);
-                auto separator_aboveAddAndRemove = popup->insertSeparator(action_addWord);
-                popup->insertActions(separator_aboveAddAndRemove, spellings_system);
-                auto separator_aboveSystemDictionarySuggestions = popup->insertSeparator(spellings_system.first());
-                popup->insertAction(separator_aboveSystemDictionarySuggestions, action_dictionarySeparatorLine);
-                auto separator_aboveDictionarySeparatorLine = popup->insertSeparator(action_dictionarySeparatorLine);
-                popup->insertActions(separator_aboveDictionarySeparatorLine, spellings_profile);
+        } else {
+            QAction* pA = nullptr;
+            auto mainConsole = mpConsole->mpHost->mpConsole;
+            if (mainConsole->isUsingSharedDictionary()) {
+                pA = new QAction(tr("no suggestions (shared)",
+                                            // Intentional comment
+                                            "used when the command spelling checker using the dictionary shared between profile has no words to suggest"));
             } else {
-                popup->insertActions(separator_aboveStandardMenu, spellings_system);
+                pA = new QAction(tr("no suggestions (profile)",
+                                            // Intentional comment
+                                            "used when the command spelling checker using the profile's own dictionary has no words to suggest"));
             }
-            // else the word is in the dictionary - in either ca`se show the context
-            // menu - either the one with the prefixed spellings, or the standard
-            // one:
+            pA->setEnabled(false);
+            spellings_profile << pA;
+        }
+    }
+
+    /*
+    * Build up the extra context menu items from the BOTTOM up, so that
+    * the top of the context menu looks like:
+    *
+    * profile dictionary suggestions
+    * --------- separator_aboveDictionarySeparatorLine
+    * \/ System dictionary suggestions /\ Profile  <== Text
+    * --------- separator_aboveSystemDictionarySuggestions
+    * system dictionary suggestions
+    * --------- separator_aboveAddAndRemove
+    * Add word action
+    * Remove word action
+    * --------- separator_aboveStandardMenu
+    *
+    * The insertAction[s](...)/(Separator(...)) insert their things
+    * second argument (or generated by themself) before the first (or
+    * only) argument given.
```

Comment:
This might clarify it a bit more - I recently revisited this area (to add controls for the PR for saving the command line history #6767) and it took me a while to get my head around the code that builds the context menu... :astonished: 
```suggestion
    * The insertAction[s](...)/(Separator(...)) insert their things
    * second argument (or generated by themself) before the first (or
    * only) argument given; they return a pointer to the item they
    * and we need that to pass into the next one we create - which is
    * why we are repeatedly capturing those pointers into `auto`
    * typed variables.
```

Reply:
I don't agree that longer is always clearer, also you introduce weird phrasing like:
"they return a pointer to the item they and we need that"

```suggestion
    * The following "insertAction", "insertActions" and "insertSeparator" need as
    * their first argument a pointer to the item after which they should insert.
    * They also return a pointer to the item they.
    * Thus we capture those pointers to give to the next functions repeatedly.
```